docs: add scalability guidance (#1413)
PowerGridModel/power-grid-model: docs/user_manual/performance-guide.md

To achieve that high performance, several optimizations are made.
To use those optimizations to the fullest, we recommend our users to follow the following guidelines.

## Scalability

The calculation workload generally depends on the number of connected grid elements and on the network topology.

> For radial distribution grids, most calculation steps scale close to linearly with grid size.

Reviewer: ```suggestion
For radial distribution grids, all calculation steps scale (amortized) linearly with grid size.
```

It's probably also good to explicitly state that it is `O(n_nodes + n_branches + ...)`

> Meshed grids require more coupling between buses and branches, so some steps can scale less favorably.

Reviewer: ```suggestion
Meshed grids require more coupling between buses and branches and will result in quadratic scaling in the largest loop in the mesh (worst case: `O(n_nodes * n_branches)`)
```

In both cases, runtime and memory usage are also affected by the number of batch scenarios, the amount of update data per
scenario, and whether the topology can be reused between scenarios.

> 

Reviewer: Since someone reading this is interested in scalabiltity, and we've already done the full analysis in https://github.com/orgs/PowerGridModel/discussions/24 , please:

* explicitly provide the scalings (i.e. it should be a summary/conclusion of https://github.com/orgs/PowerGridModel/discussions/24)
* explicitly refer to https://github.com/orgs/PowerGridModel/discussions/24 for reference + extensive details

> ```{note}

Reviewer: this note should not be part of this section; please restore the original order
